internal/dot/graph.go

```diff
@@ -20,53 +20,166 @@
 
 package dot
 
-import "fmt"
+import (
+	"fmt"
+	"reflect"
+)
 
 // Ctor encodes a constructor provided to the container for the DOT graph.
 type Ctor struct {
-	Name    string
-	Package string
-	File    string
-	Line    int
-	Params  []*Node
-	Results []*Node
+	Name       string
+	Package    string
+	File       string
+	Line       int
+	Params     []*Param
+	GroupParam []*Group
+	Results    []*Result
+}
+
+// Param is a parameter node in the graph.
+type Param struct {
+	*Node
+	Optional bool
+}
+
+// Result is a result node in the graph.
+type Result struct {
+	*Node
+	GroupIndex int
+}
+
+// Group is a group node in the graph.
+type Group struct {
+	Type    reflect.Type
```

Comment:
nit: mention that this is the type of values in the group. Otherwise, it's easy to confuse between `foo` which was added to the group and `[]foo` which is how the values in the group are consumed.

Reply:
Added comment.